common/util/vector3.h

```diff
@@ -11,7 +11,21 @@ class Vector3
 
   Vector3();
 
-  Vector3(float r, float s, float t);
+  Vector3(const float r, const float s, const float t);
+
+  Vector3(const float vect[3]);
+
+  const float getX();
```

Comment:
I get an error with warnings turned on.
```
CS5500/common/util/vector3.h:18:20: error: type qualifiers ignored on function return type [-Werror=ignored-qualifiers]
   const float getX();
                    ^
```
Try putting the `const` on the right side of the function declaration.